ai-analysis/test-08-12-25/add-custom-page-CORRECTED.md

```diff
@@ -0,0 +1,600 @@
+        </service>
+    </services>
+</container>
+```
+
+:::
+
+::: info
+**Service Registration Notes**
+
+- **Page Loader**: Must be `public="true"` to be injected into controller
+- **Controller**: Must have `controller.service_arguments` tag for Symfony routing
+- **`setContainer` call**: Required for `StorefrontController` to access services like `renderStorefront()`
+:::
+
+### Step 6: Create the Template
+
+Create a simple template to display your page.
+
+::: code-group
+
+```twig [src/Resources/views/storefront/page/example/index.html.twig]
+{% sw_extends '@Storefront/storefront/base.html.twig' %}
+
+{% block base_content %}
+    <div class="container">
+        <h1>{{ page.exampleData.title }}</h1>
+        <p>{{ page.exampleData.description }}</p>
+
+        {# Access meta information from base page #}
+        <p>Page Title: {{ page.metaInformation.metaTitle }}</p>
+    </div>
+{% endblock %}
+```
+
+:::
+
+## Testing Your Page
+
+After implementation:
+
+1. **Clear cache**: `bin/console cache:clear`
+2. **Visit your page**: Navigate to `https://your-shop.com/example-page`
+3. **Check for errors**: Monitor the Symfony profiler and logs
+4. **Test event dispatching**: Add an event subscriber to verify your `ExamplePageLoadedEvent` is fired
+
+### Example Event Subscriber (for testing)
+
+::: code-group
+
+```php [src/Subscriber/ExamplePageSubscriber.php]
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Subscriber;
+
+use Swag\BasicExample\Storefront\Page\Example\ExamplePageLoadedEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ExamplePageSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ExamplePageLoadedEvent::class => 'onExamplePageLoaded',
+        ];
+    }
+
+    public function onExamplePageLoaded(ExamplePageLoadedEvent $event): void
+    {
+        // Add custom data or modify the page
+        $page = $event->getPage();
+        
+        // Example: Add additional data
+        $additionalData = $page->getExampleData();
+        $additionalData->assign(['addedBySubscriber' => true]);
+    }
+}
+```
+
+:::
+
+Register the subscriber:
+
+```xml
+<service id="Swag\BasicExample\Subscriber\ExamplePageSubscriber">
+    <tag name="kernel.event_subscriber"/>
+</service>
+```
+
+## Common Pitfalls
+
+### ❌ Using Repositories Directly
+
+```php
+// DON'T DO THIS in page loader or controller
+public function load(Request $request, SalesChannelContext $context): ExamplePage
+{
+    $products = $this->productRepository->search($criteria, $context); // ❌ WRONG
+    // ...
+}
+```
+
+### ✅ Use Store API Routes Instead
+
+```php
+// DO THIS instead
+public function load(Request $request, SalesChannelContext $context): ExamplePage
+{
+    $products = $this->productListingRoute->load($request, $context); // ✅ CORRECT
+    // ...
+}
+```
+
+### ❌ Forgetting to Dispatch Event
```

Comment:
this is not a major issue in plugins, but it is important and the best practice for core routes, so those are extendable by other plugins, custom routes in custom plugins in general don't need to be extendable at all